pkg/kv/postgres/migrate.go

```diff
@@ -0,0 +1,186 @@
+package postgres
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/jackc/pgx/v4"
+
+	"github.com/treeverse/lakefs/pkg/kv"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/treeverse/lakefs/pkg/db/params"
+	"github.com/treeverse/lakefs/pkg/logging"
+)
+
+type MigrateFunc func(ctx context.Context, db *pgxpool.Pool, writer io.Writer) error
+
+var (
+	kvPkgs     = make(map[string]pkgMigrate)
+	registerMu sync.RWMutex
+)
+
+type pkgMigrate struct {
+	Func   MigrateFunc
+	Tables []string
+}
+
+func timeTrack(start time.Time, logger logging.Logger, name string) {
+	elapsed := time.Since(start)
+	logger.Info(fmt.Sprintf("%s took %s", name, elapsed))
+}
```

Comment:
1. our logging package reports the calling func - in this case, I would not encapsulate the call to this func as defer helps you, we will log the time took as part of a complete operation and not a failure (if you like that - sometimes you don't)
2. logger.Infof preferred in this case